addons/functions/functions/fnc_spawnTrench.sqf

```diff
@@ -0,0 +1,42 @@
+#include "script_component.hpp"
+/*
+* Author: nomisum
+* Spawns a Trench local to Server
+*
+* Arguments:
+* 0: TrenchClass <STRING>
+* 1: PosDiff <NUMBER 0-1>
+* 2: Pos <ARRAY>
+* 3: VecDirAndUp <NESTED ARRAY>
+*
+* Return Value:
+* None
+*
+* Example:
+* [TrenchObj, 0.5] call grad_trenches_functions_fnc_setTrenchProgress
+*
+* Public: No
+*/
+
+params ["_trenchClass", "_posDiff", "_pos", "_vecDirAndUp"];
+
+//Create a new trench, that is globally visible
+private _trench = createVehicle [_trenchClass, [0,0,0], [], 0, "CAN_COLLIDE"];
+private _digTime = missionNamespace getVariable [getText (configFile >> "CfgVehicles" >> _trenchClass >>"ace_trenches_diggingDuration"), 20];
+
+_trench setVariable [QGVAR(diggingSteps), (_posDiff/(_digTime*10)), true];
+if (GVAR(createTrenchMarker)) then {[_trench, side group _unit] call FUNC(createTrenchMarker)};
```

Comment:
nit pick: `side` can take a unit as parameter, no need to go via `group`.

Uhm. Speaking of which… `_unit` is not defined here, is it :rofl:

Reply:
fixed in 9978ad27836f80854cad36f6e083c694291c5c5f